app/lib/shared/email.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+final _nameEmailRegExp = new RegExp(r'^(.*)<(.+@.+)>$');
+final _defaultFrom = new EmailAddress('Pub Site Admin', 'pub@dartlang.org');
+
+/// Represents a parsed e-mail address.
+class EmailAddress {
+  final String name;
+  final String email;
+
+  EmailAddress(this.name, this.email);
+
+  factory EmailAddress.parse(String value) {
+    value = value.trim();
+    String name = value;
+    String email;
+
+    final match = _nameEmailRegExp.matchAsPrefix(value);
```

Comment:
Couldn't we propose this parse string to email, and reflow body text as static method in the `mailer` package?

Or at-least use the types defined in the mailer package, that way we don't have to wrap back and forth..?

Reply:
`EmailAddress` (previously: `Author`) is something we use in our code outside of the scope of email. I wouldn't rely on a third-party package to provide the data structure for it. Besides: I haven't seen mailer to parse the string, it only has a structure to hold it.

`EmailMessage` on the other hand could be replaced.